src/vehicle.cpp

```diff
@@ -3572,6 +3618,63 @@ void vehicle::power_parts()
         }
     }
 
+    if( has_part( "BOILER", true ) ) {
```

Comment:
If you want to have boiler cached, use `VPFLAG_BOILER` instead of `"BOILER"` here. It's faster.